### PR TITLE
[Merged by Bors] - Make sponsors section responsive

### PR DESCRIPTION
**Pull request by doup**
- This PR updates the sponsors section and makes them responsive (the current implementation breaks at mobile widths). 
- `sponsors` "component" added
- SCSS vars grouped by category
- Breakpoints SCSS vars added

**🔴 BEFORE:**

https://user-images.githubusercontent.com/188612/161599494-a3f7c086-7480-4372-8588-25a4c7304a92.mp4

**🟢 AFTER:**

image

https://user-images.githubusercontent.com/188612/161599521-eb0ed444-dc5a-4380-88ea-cd8dcd5760df.mp4

### CODE REVIEW

**Review by IceSentry — APPROVED** [review]
LGTM

**Comment by alice-i-cecile** [comment]
I'm on board with this direction, but I'm not confident in my ability to review the changes myself :)

File: templates/index.html [hunk 1]
```diff
@@ -187,21 +187,41 @@ <h2 class="feature-title">Free and Open Source</h2>
...
-        <div class="sponsors-section">
-            <a href="https://agileperception.com/"><img src="assets/agile_perception.png" class="silver-sponsor-image" alt="Agile Perception logo"/></a>
+        <div class="sponsors sponsors--platinum">
+            <h2 class="sponsors__title">Platinum Sponsors</h2>
+            <div class="sponsors__content">
+                <a class="sponsors__link" href="https://www.embark-studios.com">
+                    <img class="sponsors__logo" src="assets/embark.png" alt="Embark Studios logo"/>
+                </a>
+                <a class="sponsors__link" href="https://legionlabs.com">
+                    <img class="sponsors__logo" src="assets/LegionLabs.png" alt="Legion Labs logo"/>
```
Comment (cart) [inline templates/index.html:197]: Legion Labs had a manual scale override to make it closer in size to the embark logo. We should either re-add the override or scale the image itself to ensure we aren't giving unfair prominence to Legion Labs.

File: sass/components/_sponsors.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    &:not(:last-child) {
+        margin-bottom: 56px;
+    }
+
+    &__title {
+        font-size: 2.2rem;
+        margin: 0 0 24px;
+    }
+
+    &__content {
```
Comment (cart) [inline sass/components/_sponsors.scss:15]: Haha I've been webdeving for a long time and I've never used this pattern. Whats going on here / whats the name of this feature?

File: templates/index.html (hunk 1 above)
Comment (cart) [reply templates/index.html:197]: Works for me!

File: sass/components/_sponsors.scss (hunk 2 above)
Comment (cart) [reply sass/components/_sponsors.scss:15]: Ahh the `&` is a sass feature.

Comment (IceSentry) [reply sass/components/_sponsors.scss:15]: Oh, yeah, I wasn't sure what you were referring to.

File: templates/index.html (hunk 1 above)
Comment (doup) [reply templates/index.html:197]: I've added a bigger Embark logo, and now it takes the 100px height. The size difference should be less noticeable (it's still not totally balanced though). Anyway, this should be revisited at some time, it's a little bit tricky to keep them balanced, reponsive, while maintaining the aspect ratio and with good spacing between them… 😅 

Oh, btw, I've found high-quality embark logos on the Arc Riders press kit: https://drive.google.com/drive/folders/1u8D3kF0GKKuouXeOf7qU0a6ReNSOv0hd

Comment (doup) [reply templates/index.html:197]: OK, I've also tweaked the LegionLabs logo. So now both of them have 200px height and same "x-height" (lowercase "x" height). **Good enough?** :P

image

image

Comment (alice-i-cecile) [reply templates/index.html:197]: Good enough!

File: sass/components/_sponsors.scss (hunk 2 above)
Comment (cart) [reply sass/components/_sponsors.scss:15]: It was a combination of both things. I've havent used bem before either. Thanks for the info!

File: templates/index.html (hunk 1 above)
Comment (cart) [reply templates/index.html:197]: Agreed!

**Pull request closed without merging**